### PR TITLE
75505 scenario mapping implementation for new cell & cancelreplace cell

### PR DESCRIPTION
**Pull request by Nadeem14456**
This PR includes below functionality.
 - PBI#75505 - Scenario identification and mapping.
 - PBI#76683 - New cell rules and required attributes
 - PBI#76750 - Cancelled & Replaced rules and required attributes
 - Unit Test cases

### CODE REVIEW

File: src/UKHO.ERPFacade.API/appsettings.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "ConnectionString": ""
   },
   "SapConfiguration": {
-    "BaseAddress": "",
+    "BaseAddress": "https://localhost:60032/",
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.API/appsettings.json:38]:
```suggestion
    "BaseAddress": "",
```

File: src/UKHO.ERPFacade.Common/Logging/EventIds.cs [hunk 2]
```diff
@@ -94,5 +94,29 @@ public enum EventIds
...
+        /// </summary>
+        SapActionCreated = 940027,
+        /// <summary>
+        /// 940028 - Identifying Scenario Started
+        /// </summary>
+        IdentifyScenarioStarted = 940028,
+        /// <summary>
+        /// 940029 - Scenario Identified
+        /// </summary>
+        ScenarioIdentified = 940029,
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.Common/Logging/EventIds.cs:120]:
```suggestion
        ScenarioIdentified = 940029
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+            {
+                itemNode.AppendChild(itemAttribute.itemNode);
+            }
+            return itemNode;
+        }
+
+        private static string GetProdType(string prodType)
+        {
+            var parts = prodType.Split(' ').ToList();
+            return parts[1];
```
[the remark this answers is not in the document]
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs]: Suggested changes done.

File: src/UKHO.ERPFacade.API/appsettings.json (hunk 1 above)
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/appsettings.json]: Suggested changes done.

File: src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+            foreach (var node in action.Attributes.Where(x => x.Section == ProductSection))
+            {
+                XmlElement itemSubNode = soapXml.CreateElement(node.XmlNodeName);
+
+                if (node.IsRequired)
+                {
+                    if (node.XmlNodeName == ReplacedBy)
+                    {
+                        itemSubNode.InnerText = string.IsNullOrWhiteSpace(cell.ToString()) ? string.Empty : cell.ToString().Substring(0, Math.Min(250, cell.ToString().Length));
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs:182]: Can we make one method to get this value and use it wherever needed as same is repeated couple of times.

File: src/UKHO.ERPFacade.WebJob/ErpFacadeWebJob.cs [hunk 5]
```diff
@@ -33,9 +33,9 @@ public void Start()
             try
             {
                 //Code to monitor the table records.
-                _logger.LogInformation(EventIds.WebjobProcessEventStarted.ToEventId(), "Webjob started for the processing the incomplete transactions.");
+                _logger.LogInformation(EventIds.WebjobProcessEventStarted.ToEventId(), "Webjob started for monitoring the callbacks from SAP.");
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.WebJob/ErpFacadeWebJob.cs:36]:
```suggestion
                _logger.LogInformation(EventIds.WebjobProcessEventStarted.ToEventId(), "Webjob started the monitoring of callbacks from SAP.");
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs (hunk 4 above)
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Helpers/SapMessageBuilder.cs:182]: done.

**Review by harshal11869 — APPROVED** [review]
LGTM

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Pull request merged**